### DISCUSSION TITLE
Example of internal framework build with Vike

### DISCUSSION THREAD

**Discussion by paulogr** [vikejs/vike#1357] [Discussion]
Does anyone have an example of an internal framework build with Vike?

I see @brillout share some public repos with examples of public frameworks, but I'm intend to build a internal framework to use at my own company.

Thank you!

**Comment by paulogr** [comment]
I'm trying to achieve this folder structure

![Screenshot from 2023-12-14 10-33-10](https://github.com/vikejs/vike/assets/30919/1a92f252-d30f-4b85-b7f4-7747e16a0e8f)

**Comment by brillout** [comment]
We have a couple of sponsors who do that and we've been helping them with setting up their internal frameworks. Is [sponsoring](https://github.com/sponsors/vikejs) something your company would be up for? Building an advanced framework isn't completely trivial (regardless of Vike).

**Reply by paulogr** [reply]
Thank for your answer @brillout, but not for now.

I've just had my first contact with the framework and I was catched by the idea to create "my own framework".

I just would like to understand how this works for now, just saw the "extensions" property exposed by the plugin but it seens like you always have to export a package name but it wasn´t what I am looking for.

Thanks, and as soon Vike be part of our tech stack I would be happy to sponsor it.

**Reply by brillout** [reply]
The `extends` keyword (see docs) is new and you don't have to use it. Instead, you can move code to an npm package and then use that package in user land (that's what the sponsors are doing).

I think, at this point, the best is that you try and see how it goes. If you hit any blocker caused by Vike, feel free to let us know. Although we're quite busy so I'll ask you to [dig a *lot*](https://brillout.github.io/rules/#try) before reaching out.

Building your own framework is powerful and I clearly encourage folks to do it, but it isn't as trivial as building a simple Nuxt app.
